### ISSUE TITLE
8wtm_auto_switch : syntax error in expression

### ISSUE THREAD

**Issue by cartesien** [papampi/nvOC_by_fullzero#102]
"/home/m1/NVOC/mining/8wtm_auto_switch: line 135: ((: 100
94: syntax error in expression (error token is "94")
Mining ETH (ETHASH) for 0 minutes, continue mining ETH for 1195 more seconds

**Comment by LuKePicci** [comment]
I need some more info about your 8wtm_auto_switch file version and WTM_top_coin file contents (if exist).

**Comment by cartesien** [comment]
I think the problem is in this expression : 
 TOP_COIN=$(sed 's/:[0-9]*//' < ${NVOC}/WTM_top_coin)

    if [[ $CURRENT_COIN != $TOP_COIN ]]; then

      #If miner running time is more than given time or profit is more than given value switch the coin
      FORCE_SWITCH=0
      CURRENT_COIN_PROFIT=$(grep "$COIN"  < ${NVOC}/WTM_current_profit | sed 's/.*://')
      TOP_COIN_PROFIT=$(sed 's/.*://' < ${NVOC}/WTM_top_coin)

      if (( TOP_COIN_PROFIT - CURRENT_COIN_PROFIT >= WTM_FORCE_SWITCH ))
      then
....

**Comment by LuKePicci** [comment]
Yeah, but I can't catch why this gives you that error. If `${NVOC}/WTM_top_coin` is not a valid or existing file it should give you a file not found error, if it is empty it also does not give a syntax error. Could you tell me what is inside WTM_top_coin when it give you this error? Maybe an issue with coins having a "-" in their name.

Edit: please, post also your WTM_current_profit file content.

**Comment by cartesien** [comment]
This errors randomly occur : 
Copy of several hours in WTM guake tab : 

Mining ETH for 90 minutes
Minimum mining time passed, checking for new coins with 7% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5335.56

 BTC PRICE: 5335.56 EUR 
 
ETH 7.53 EUR
NICE_ETHASH 7.24 EUR
ETC 7.13 EUR
ZEN 6.91 EUR
NICE_EQUIHASH 6.55 EUR
KMD 6.41 EUR
ZCL 6.27 EUR
ZEC 5.04 EUR
NICE_NEOSCRYPT 4.93 EUR
FTC 4.85 EUR

 New profits : 

ETH: 100 %
NICE_ETHASH: 96 %
ETC: 95 %
ZEN: 92 %
NICE_EQUIHASH: 87 %
KMD: 85 %
ZCL: 83 %
ZEC: 67 %
NICE_NEOSCRYPT: 65 %
FTC: 64 %
Tue Jun 26 01:11:16 CEST 2018 - Same Coin on Top, Continue mining ETH
24 Hour Average Revenue:  7.71073 EUR
Check again in 900 seconds


Mining ETH for 105 minutes
Minimum mining time passed, checking for new coins with 7% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5335.3875714

 BTC PRICE: 5335.3875714 EUR 
 
ZCL 9.06 EUR
ETH 7.49 EUR
NICE_ETHASH 7.24 EUR
ETC 7.16 EUR
NICE_EQUIHASH 6.81 EUR
ZEN 6.13 EUR
KMD 5.55 EUR
NICE_NEOSCRYPT 5.21 EUR
ZEC 5.19 EUR
FTC 4.85 EUR

 New profits : 

ZCL: 121 %
ETH: 100 %
NICE_ETHASH: 97 %
ETC: 96 %
NICE_EQUIHASH: 91 %
ZEN: 82 %
KMD: 74 %
NICE_NEOSCRYPT: 70 %
ZEC: 69 %
FTC: 65 %
/home/m1/NVOC/mining/8wtm_auto_switch: line 135: ((: 100
97: syntax error in expression (error token is "97")
Current Coin Algo: NEOSCRYPT
Current Coin: ETH
New Top Coin: ZCL
Switching Coin, restarting 3main
WTM Auto Switch Killed watchdog
Tue Jun 26 01:26:25 CEST 2018 - Mining switched from ETH to ZCL
24 Hour Average Revenue:  7.72094 EUR
Check again in 900 seconds


Mining ZCL for 15 minutes
Minimum mining time not reached, checking for new coins with 10% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5335.4

 BTC PRICE: 5335.4 EUR 
 
ETH 7.48 EUR
ETC 7.25 EUR
NICE_ETHASH 7.24 EUR
NICE_NEOSCRYPT 7.03 EUR
ZCL 6.69 EUR
NICE_EQUIHASH 6.11 EUR
ZEC 5.91 EUR
ZEN 5.31 EUR
KMD 5.19 EUR
FTC 4.99 EUR

 New profits : 

ETH: 100 %
ETC: 97 %
NICE_ETHASH: 97 %
NICE_NEOSCRYPT: 94 %
ZCL: 89 %
NICE_EQUIHASH: 82 %
ZEC: 79 %
ZEN: 71 %
KMD: 69 %
FTC: 67 %
New coin has %11 more profit, ignore timer and make the switch
Current Coin Algo: NEOSCRYPT
Current Coin: ZCL
New Top Coin: ETH
Switching Coin, restarting 3main
WTM Auto Switch Killed watchdog
Tue Jun 26 01:41:33 CEST 2018 - Mining switched from ZCL to ETH
24 Hour Average Revenue:  7.71792 EUR
Check again in 900 seconds


Mining ETH for 15 minutes
Minimum mining time not reached, checking for new coins with 10% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5337.97

 BTC PRICE: 5337.97 EUR 
 
ETH 7.50 EUR
ETC 7.26 EUR
NICE_ETHASH 7.24 EUR
ZCL 6.25 EUR
NICE_EQUIHASH 6.07 EUR
ZEC 5.98 EUR
FTC 5.53 EUR
KMD 5.18 EUR
ZEN 5.11 EUR
NICE_NEOSCRYPT 4.82 EUR

 New profits : 

ETH: 100 %
ETC: 97 %
NICE_ETHASH: 97 %
ZCL: 83 %
NICE_EQUIHASH: 81 %
ZEC: 80 %
FTC: 74 %
KMD: 69 %
ZEN: 68 %
NICE_NEOSCRYPT: 64 %
Tue Jun 26 01:56:40 CEST 2018 - Same Coin on Top, Continue mining ETH
24 Hour Average Revenue:  7.71792 EUR
Check again in 900 seconds


Mining ETH for 30 minutes
Minimum mining time passed, checking for new coins with 7% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5346.78

 BTC PRICE: 5346.78 EUR 
 
ETH 7.57 EUR
ZEC 7.39 EUR
ETC 7.28 EUR
NICE_ETHASH 7.26 EUR
FTC 6.11 EUR
ZCL 6.07 EUR
NICE_EQUIHASH 6.05 EUR
KMD 5.31 EUR
ZEN 5.11 EUR
NICE_NEOSCRYPT 4.94 EUR

 New profits : 

ETH: 100 %
ZEC: 98 %
ETC: 96 %
NICE_ETHASH: 96 %
FTC: 81 %
ZCL: 80 %
NICE_EQUIHASH: 80 %
KMD: 70 %
ZEN: 67 %
NICE_NEOSCRYPT: 65 %
Tue Jun 26 02:11:47 CEST 2018 - Same Coin on Top, Continue mining ETH
24 Hour Average Revenue:  7.71521 EUR
Check again in 900 seconds


Mining ETH for 45 minutes
Minimum mining time passed, checking for new coins with 7% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5335.2452714

 BTC PRICE: 5335.2452714 EUR 
 
ETH 7.55 EUR
ZEC 7.32 EUR
ETC 7.32 EUR
NICE_ETHASH 7.24 EUR
FTC 6.16 EUR
ZCL 6.09 EUR
NICE_EQUIHASH 6.06 EUR
KMD 5.83 EUR
NICE_NEOSCRYPT 5.36 EUR
ZEN 5.10 EUR

 New profits : 

ETH: 100 %
ZEC: 97 %
ETC: 97 %
NICE_ETHASH: 96 %
FTC: 82 %
ZCL: 81 %
NICE_EQUIHASH: 80 %
KMD: 77 %
NICE_NEOSCRYPT: 71 %
ZEN: 68 %
Tue Jun 26 02:26:54 CEST 2018 - Same Coin on Top, Continue mining ETH
24 Hour Average Revenue:  7.71177 EUR
Check again in 900 seconds


Mining ETH for 60 minutes
Minimum mining time passed, checking for new coins with 7% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5322.0755394

 BTC PRICE: 5322.0755394 EUR 
 
FTC 7.89 EUR
ETH 7.61 EUR
ETC 7.30 EUR
NICE_ETHASH 7.22 EUR
ZEC 6.93 EUR
NICE_EQUIHASH 6.14 EUR
ZCL 6.13 EUR
KMD 6.10 EUR
NICE_NEOSCRYPT 5.77 EUR
ZEN 5.09 EUR

 New profits : 

FTC: 104 %
ETH: 100 %
ETC: 96 %
NICE_ETHASH: 95 %
ZEC: 91 %
NICE_EQUIHASH: 81 %
ZCL: 80 %
KMD: 80 %
NICE_NEOSCRYPT: 76 %
ZEN: 67 %
Tue Jun 26 02:42:01 CEST 2018 - Same Coin on Top, Continue mining ETH
24 Hour Average Revenue:  7.70688 EUR
Check again in 900 seconds


Mining ETH for 75 minutes
Minimum mining time passed, checking for new coins with 7% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5316.72

 BTC PRICE: 5316.72 EUR 
 
ETH 7.62 EUR
ETC 7.27 EUR
NICE_ETHASH 7.22 EUR
ZEC 6.67 EUR
ZCL 6.53 EUR
KMD 6.32 EUR
NICE_NEOSCRYPT 6.22 EUR
NICE_EQUIHASH 6.04 EUR
ZEN 5.14 EUR
FTC 5.02 EUR

 New profits : 

ETH: 100 %
ETC: 95 %
NICE_ETHASH: 95 %
ZEC: 88 %
ZCL: 86 %
KMD: 83 %
NICE_NEOSCRYPT: 82 %
NICE_EQUIHASH: 79 %
ZEN: 68 %
FTC: 66 %
Tue Jun 26 02:57:08 CEST 2018 - Same Coin on Top, Continue mining ETH
24 Hour Average Revenue:  7.65698 EUR
Check again in 900 seconds


Mining ETH for 90 minutes
Minimum mining time passed, checking for new coins with 7% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5304.34

 BTC PRICE: 5304.34 EUR 
 
ZCL 8.68 EUR
ETH 7.60 EUR
ETC 7.24 EUR
NICE_ETHASH 7.20 EUR
NICE_EQUIHASH 6.87 EUR
ZEC 6.74 EUR
KMD 5.51 EUR
FTC 5.48 EUR
ZEN 5.12 EUR
NICE_NEOSCRYPT 4.16 EUR

 New profits : 

ZCL: 114 %
ETH: 100 %
ETC: 95 %
NICE_ETHASH: 95 %
NICE_EQUIHASH: 90 %
ZEC: 89 %
KMD: 73 %
FTC: 72 %
ZEN: 67 %
NICE_NEOSCRYPT: 55 %
/home/m1/NVOC/mining/8wtm_auto_switch: line 135: ((: 100
95: syntax error in expression (error token is "95")
Current Coin Algo: NEOSCRYPT
Current Coin: ETH
New Top Coin: ZCL
Switching Coin, restarting 3main
WTM Auto Switch Killed watchdog
Tue Jun 26 03:12:17 CEST 2018 - Mining switched from ETH to ZCL
24 Hour Average Revenue:  7.65656 EUR
Check again in 900 seconds


Mining ZCL for 15 minutes
Minimum mining time not reached, checking for new coins with 10% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5305.78

 BTC PRICE: 5305.78 EUR 
 
ETH 7.67 EUR
ETC 7.33 EUR
NICE_ETHASH 7.20 EUR
ZEC 6.57 EUR
ZCL 6.24 EUR
NICE_EQUIHASH 6.07 EUR
KMD 5.31 EUR
FTC 5.25 EUR
ZEN 5.18 EUR
NICE_NEOSCRYPT 5.02 EUR

 New profits : 

ETH: 100 %
ETC: 96 %
NICE_ETHASH: 94 %
ZEC: 86 %
ZCL: 81 %
NICE_EQUIHASH: 79 %
KMD: 69 %
FTC: 68 %
ZEN: 68 %
NICE_NEOSCRYPT: 65 %
New coin has %19 more profit, ignore timer and make the switch
Current Coin Algo: NEOSCRYPT
Current Coin: ZCL
New Top Coin: ETH
Switching Coin, restarting 3main
WTM Auto Switch Killed watchdog
Tue Jun 26 03:27:25 CEST 2018 - Mining switched from ZCL to ETH
24 Hour Average Revenue:  7.64875 EUR
Check again in 900 seconds


Mining ETH for 15 minutes
Minimum mining time not reached, checking for new coins with 10% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5320.3

 BTC PRICE: 5320.3 EUR 
 
ETH 7.66 EUR
ETC 7.36 EUR
NICE_ETHASH 7.22 EUR
ZEN 7.07 EUR
ZEC 7.06 EUR
NICE_EQUIHASH 6.04 EUR
ZCL 5.92 EUR
KMD 4.92 EUR
FTC 4.86 EUR
NICE_NEOSCRYPT 4.61 EUR

 New profits : 

ETH: 100 %
ETC: 96 %
NICE_ETHASH: 94 %
ZEN: 92 %
ZEC: 92 %
NICE_EQUIHASH: 79 %
ZCL: 77 %
KMD: 64 %
FTC: 63 %
NICE_NEOSCRYPT: 60 %
Tue Jun 26 03:42:31 CEST 2018 - Same Coin on Top, Continue mining ETH
24 Hour Average Revenue:  7.64875 EUR
Check again in 900 seconds


Mining ETH for 30 minutes
Minimum mining time passed, checking for new coins with 7% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5321.37

 BTC PRICE: 5321.37 EUR 
 
ZEN 7.73 EUR
ETH 7.65 EUR
ZEC 7.51 EUR
ETC 7.44 EUR
NICE_ETHASH 7.22 EUR
NICE_EQUIHASH 6.51 EUR
NICE_NEOSCRYPT 5.91 EUR
ZCL 5.67 EUR
KMD 5.03 EUR
FTC 5.00 EUR

 New profits : 

ZEN: 101 %
ETH: 100 %
ZEC: 98 %
ETC: 97 %
NICE_ETHASH: 94 %
NICE_EQUIHASH: 85 %
NICE_NEOSCRYPT: 77 %
ZCL: 74 %
KMD: 66 %
FTC: 65 %
Tue Jun 26 03:57:38 CEST 2018 - Same Coin on Top, Continue mining ETH
24 Hour Average Revenue:  7.64229 EUR
Check again in 900 seconds


Mining ETH for 45 minutes
Minimum mining time passed, checking for new coins with 7% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5322.3983398

 BTC PRICE: 5322.3983398 EUR 
 
ZEC 8.23 EUR
ETH 7.62 EUR
ETC 7.43 EUR
NICE_ETHASH 7.22 EUR
ZEN 6.51 EUR
NICE_EQUIHASH 5.94 EUR
NICE_NEOSCRYPT 5.73 EUR
ZCL 5.55 EUR
KMD 5.24 EUR
FTC 4.17 EUR

 New profits : 

ZEC: 108 %
ETH: 100 %
ETC: 98 %
NICE_ETHASH: 95 %
ZEN: 85 %
NICE_EQUIHASH: 78 %
NICE_NEOSCRYPT: 75 %
ZCL: 73 %
KMD: 69 %
FTC: 55 %
/home/m1/NVOC/mining/8wtm_auto_switch: line 135: ((: 100
95: syntax error in expression (error token is "95")
Current Coin Algo: NEOSCRYPT
Current Coin: ETH
New Top Coin: ZEC
Switching Coin, restarting 3main
WTM Auto Switch Killed watchdog
Tue Jun 26 04:12:47 CEST 2018 - Mining switched from ETH to ZEC
24 Hour Average Revenue:  7.64698 EUR
Check again in 900 seconds


Mining ZEC for 15 minutes
Minimum mining time not reached, checking for new coins with 10% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5325.45

 BTC PRICE: 5325.45 EUR 
 
ZEC 8.71 EUR
ETH 7.63 EUR
ETC 7.43 EUR
NICE_ETHASH 7.23 EUR
KMD 6.34 EUR
ZCL 5.99 EUR
NICE_EQUIHASH 5.96 EUR
ZEN 5.94 EUR
NICE_NEOSCRYPT 5.06 EUR
FTC 3.84 EUR

 New profits : 

ZEC: 114 %
ETH: 100 %
ETC: 97 %
NICE_ETHASH: 95 %
KMD: 83 %
ZCL: 79 %
NICE_EQUIHASH: 78 %
ZEN: 78 %
NICE_NEOSCRYPT: 66 %
FTC: 50 %
Tue Jun 26 04:27:53 CEST 2018 - Same Coin on Top, Continue mining ZEC
24 Hour Average Revenue:  7.64698 EUR
Check again in 900 seconds


Mining ZEC for 30 minutes
Minimum mining time passed, checking for new coins with 7% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5323.06

 BTC PRICE: 5323.06 EUR 
 
ZEC 8.94 EUR
ETH 7.65 EUR
ETC 7.45 EUR
NICE_ETHASH 7.22 EUR
ZCL 7.19 EUR
NICE_EQUIHASH 6.25 EUR
KMD 6.25 EUR
ZEN 5.93 EUR
NICE_NEOSCRYPT 5.42 EUR
FTC 5.20 EUR

 New profits : 

ZEC: 117 %
ETH: 100 %
ETC: 97 %
NICE_ETHASH: 94 %
ZCL: 94 %
NICE_EQUIHASH: 82 %
KMD: 82 %
ZEN: 78 %
NICE_NEOSCRYPT: 71 %
FTC: 68 %
Tue Jun 26 04:43:00 CEST 2018 - Same Coin on Top, Continue mining ZEC
24 Hour Average Revenue:  7.65667 EUR
Check again in 900 seconds


Mining ZEC for 45 minutes
Minimum mining time passed, checking for new coins with 7% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5323.88

 BTC PRICE: 5323.88 EUR 
 
ZEC 8.76 EUR
ETH 7.73 EUR
ETC 7.46 EUR
NICE_ETHASH 7.09 EUR
ZCL 6.09 EUR
ZEN 5.89 EUR
NICE_EQUIHASH 5.82 EUR
NICE_NEOSCRYPT 5.75 EUR
KMD 5.59 EUR
FTC 5.22 EUR

 New profits : 

ZEC: 113 %
ETH: 100 %
ETC: 96 %
NICE_ETHASH: 92 %
ZCL: 79 %
ZEN: 76 %
NICE_EQUIHASH: 75 %
NICE_NEOSCRYPT: 74 %
KMD: 72 %
FTC: 68 %
Tue Jun 26 04:58:07 CEST 2018 - Same Coin on Top, Continue mining ZEC
24 Hour Average Revenue:  7.67313 EUR
Check again in 900 seconds


Mining ZEC for 60 minutes
Minimum mining time passed, checking for new coins with 7% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5323.4092701

 BTC PRICE: 5323.4092701 EUR 
 
ZEC 8.47 EUR
ETH 7.70 EUR
ETC 7.49 EUR
NICE_ETHASH 7.22 EUR
ZEN 5.99 EUR
ZCL 5.72 EUR
NICE_EQUIHASH 5.59 EUR
FTC 5.06 EUR
NICE_NEOSCRYPT 4.66 EUR
KMD 4.53 EUR

 New profits : 

ZEC: 110 %
ETH: 100 %
ETC: 97 %
NICE_ETHASH: 94 %
ZEN: 78 %
ZCL: 74 %
NICE_EQUIHASH: 73 %
FTC: 66 %
NICE_NEOSCRYPT: 60 %
KMD: 59 %
Tue Jun 26 05:13:14 CEST 2018 - Same Coin on Top, Continue mining ZEC
24 Hour Average Revenue:  7.68375 EUR
Check again in 900 seconds


Mining ZEC for 75 minutes
Minimum mining time passed, checking for new coins with 7% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5319.83

 BTC PRICE: 5319.83 EUR 
 
ZEC 8.31 EUR
ETH 7.70 EUR
ETC 7.53 EUR
NICE_ETHASH 7.22 EUR
ZCL 5.89 EUR
ZEN 5.85 EUR
NICE_EQUIHASH 5.17 EUR
FTC 5.15 EUR
NICE_NEOSCRYPT 4.64 EUR
KMD 4.27 EUR

 New profits : 

ZEC: 108 %
ETH: 100 %
ETC: 98 %
NICE_ETHASH: 94 %
ZCL: 76 %
ZEN: 76 %
NICE_EQUIHASH: 67 %
FTC: 67 %
NICE_NEOSCRYPT: 60 %
KMD: 55 %
Tue Jun 26 05:28:21 CEST 2018 - Same Coin on Top, Continue mining ZEC
24 Hour Average Revenue:  7.69531 EUR
Check again in 900 seconds


Mining ZEC for 90 minutes
Minimum mining time passed, checking for new coins with 7% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5323.67

 BTC PRICE: 5323.67 EUR 
 
ZEC 8.16 EUR
ETH 7.75 EUR
ETC 7.53 EUR
NICE_ETHASH 7.23 EUR
ZEN 5.95 EUR
NICE_EQUIHASH 5.71 EUR
ZCL 5.55 EUR
NICE_NEOSCRYPT 4.85 EUR
FTC 4.53 EUR
KMD 4.06 EUR

 New profits : 

ZEC: 105 %
ETH: 100 %
ETC: 97 %
NICE_ETHASH: 93 %
ZEN: 77 %
NICE_EQUIHASH: 74 %
ZCL: 72 %
NICE_NEOSCRYPT: 63 %
FTC: 58 %
KMD: 52 %
Tue Jun 26 05:43:28 CEST 2018 - Same Coin on Top, Continue mining ZEC
24 Hour Average Revenue:  7.70146 EUR
Check again in 900 seconds


Mining ZEC for 105 minutes
Minimum mining time passed, checking for new coins with 7% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5331.28

 BTC PRICE: 5331.28 EUR 
 
ZEC 8.42 EUR
ETH 7.67 EUR
ETC 7.57 EUR
NICE_ETHASH 7.24 EUR
ZEN 6.62 EUR
NICE_EQUIHASH 6.09 EUR
ZCL 5.88 EUR
FTC 4.80 EUR
NICE_NEOSCRYPT 4.76 EUR
KMD 4.21 EUR

 New profits : 

ZEC: 110 %
ETH: 100 %
ETC: 99 %
NICE_ETHASH: 94 %
ZEN: 86 %
NICE_EQUIHASH: 79 %
ZCL: 77 %
FTC: 63 %
NICE_NEOSCRYPT: 62 %
KMD: 55 %
Tue Jun 26 05:58:34 CEST 2018 - Same Coin on Top, Continue mining ZEC
24 Hour Average Revenue:  7.70969 EUR
Check again in 900 seconds


Mining ZEC for 120 minutes
Minimum mining time passed, checking for new coins with 7% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5320.76

 BTC PRICE: 5320.76 EUR 
 
ZEC 9.07 EUR
ETH 7.67 EUR
ETC 7.59 EUR
NICE_ETHASH 7.08 EUR
ZEN 7.00 EUR
NICE_EQUIHASH 6.07 EUR
ZCL 5.95 EUR
NICE_NEOSCRYPT 4.78 EUR
FTC 4.47 EUR
KMD 4.43 EUR

 New profits : 

ZEC: 118 %
ETH: 100 %
ETC: 99 %
NICE_ETHASH: 92 %
ZEN: 91 %
NICE_EQUIHASH: 79 %
ZCL: 78 %
NICE_NEOSCRYPT: 62 %
FTC: 58 %
KMD: 58 %
Tue Jun 26 06:13:41 CEST 2018 - Same Coin on Top, Continue mining ZEC
24 Hour Average Revenue:  7.71604 EUR
Check again in 900 seconds


Mining ZEC for 135 minutes
Minimum mining time passed, checking for new coins with 7% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5316.5275279

 BTC PRICE: 5316.5275279 EUR 
 
ZEC 9.24 EUR
ETH 7.70 EUR
ETC 7.56 EUR
NICE_ETHASH 7.49 EUR
ZCL 7.12 EUR
ZEN 6.17 EUR
NICE_EQUIHASH 5.98 EUR
NICE_NEOSCRYPT 5.15 EUR
KMD 4.96 EUR
FTC 4.30 EUR

 New profits : 

ZEC: 120 %
ETH: 100 %
ETC: 98 %
NICE_ETHASH: 97 %
ZCL: 92 %
ZEN: 80 %
NICE_EQUIHASH: 78 %
NICE_NEOSCRYPT: 67 %
KMD: 64 %
FTC: 56 %
Tue Jun 26 06:28:48 CEST 2018 - Same Coin on Top, Continue mining ZEC
24 Hour Average Revenue:  7.73344 EUR
Check again in 900 seconds


Mining ZEC for 151 minutes
Minimum mining time passed, checking for new coins with 7% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5329.24

 BTC PRICE: 5329.24 EUR 
 
ZEC 8.95 EUR
ZCL 8.87 EUR
ETH 7.72 EUR
ETC 7.62 EUR
NICE_ETHASH 7.23 EUR
NICE_EQUIHASH 7.09 EUR
ZEN 6.15 EUR
KMD 5.38 EUR
FTC 4.84 EUR
NICE_NEOSCRYPT 4.83 EUR

 New profits : 

ZEC: 116 %
ZCL: 115 %
ETH: 100 %
ETC: 99 %
NICE_ETHASH: 94 %
NICE_EQUIHASH: 92 %
ZEN: 80 %
KMD: 70 %
FTC: 63 %
NICE_NEOSCRYPT: 63 %
Tue Jun 26 06:43:55 CEST 2018 - Same Coin on Top, Continue mining ZEC
24 Hour Average Revenue:  7.74865 EUR
Check again in 900 seconds


Mining ZEC for 166 minutes
Minimum mining time passed, checking for new coins with 7% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5331.6

 BTC PRICE: 5331.6 EUR 
 
ZEC 8.43 EUR
ETH 7.68 EUR
ETC 7.64 EUR
NICE_ETHASH 7.24 EUR
ZCL 6.41 EUR
NICE_EQUIHASH 6.06 EUR
ZEN 5.67 EUR
NICE_NEOSCRYPT 5.21 EUR
FTC 4.82 EUR
KMD 4.37 EUR

 New profits : 

ZEC: 110 %
ETH: 100 %
ETC: 99 %
NICE_ETHASH: 94 %
ZCL: 83 %
NICE_EQUIHASH: 79 %
ZEN: 74 %
NICE_NEOSCRYPT: 68 %
FTC: 63 %
KMD: 57 %
Tue Jun 26 06:59:02 CEST 2018 - Same Coin on Top, Continue mining ZEC
24 Hour Average Revenue:  7.76469 EUR
Check again in 900 seconds


Mining ZEC for 181 minutes
Minimum mining time passed, checking for new coins with 7% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5344.25

 BTC PRICE: 5344.25 EUR 
 
ZEC 8.65 EUR
ETH 7.74 EUR
ETC 7.67 EUR
NICE_ETHASH 7.25 EUR
ZCL 6.04 EUR
ZEN 5.89 EUR
NICE_EQUIHASH 5.87 EUR
FTC 4.95 EUR
NICE_NEOSCRYPT 4.91 EUR
KMD 4.04 EUR

 New profits : 

ZEC: 112 %
ETH: 100 %
ETC: 99 %
NICE_ETHASH: 94 %
ZCL: 78 %
ZEN: 76 %
NICE_EQUIHASH: 76 %
FTC: 64 %
NICE_NEOSCRYPT: 63 %
KMD: 52 %
Tue Jun 26 07:14:09 CEST 2018 - Same Coin on Top, Continue mining ZEC
24 Hour Average Revenue:  7.77229 EUR
Check again in 900 seconds


Mining ZEC for 196 minutes
Minimum mining time passed, checking for new coins with 7% more profit
Can not get data from coindesk.com
Trying to get data from coinbase.com
Can not get data from coinbase.com
Trying to get data from blockchain.info
5345.64

 BTC PRICE: 5345.64 EUR 
 
ZEC 8.40 EUR
ETH 7.74 EUR
ETC 7.67 EUR
NICE_ETHASH 7.26 EUR
ZCL 6.04 EUR
NICE_EQUIHASH 5.90 EUR
ZEN 5.82 EUR
FTC 5.46 EUR
NICE_NEOSCRYPT 4.71 EUR
KMD 3.76 EUR

 New profits : 

ZEC: 108 %
ETH: 100 %
ETC: 99 %
NICE_ETHASH: 94 %
ZCL: 78 %
NICE_EQUIHASH: 76 %
ZEN: 75 %
FTC: 71 %
NICE_NEOSCRYPT: 61 %
KMD: 49 %
Tue Jun 26 07:29:16 CEST 2018 - Same Coin on Top, Continue mining ZEC
24 Hour Average Revenue:  7.78531 EUR
Check again in 900 seconds

Is it possible that "\n" introduced in WTM_SWITCHER python script by me to better readings can cause theses aleatory erros ?

**Comment by LuKePicci** [comment]
Maybe, could you show me your mod? I was thinking there's a problem with line endings, but I can't explain why it doesn't give error all the times. When it sayts 'error token is: X", X is always equal to the first coin profitability in the list which has _ in its name.

**Comment by cartesien** [comment]
I think the problem come from WTM website uses dash instead of Undercores in their Listing comparisons.

I saw this problem in the past. But some guys wanted to "normalize" this by replacing dashs with underscores because according to them, dash will create some problems. I notice that this make prolems now since theses modifications.

I succeeded to replace this in the scripts in the past, but now all scripts are more complex to understand for me. I could try, but i have not enough time for that.

Rentability had massively décreased since this prévious post. And even if BTC ahve the same price today, all altcoins became an increasing active cost in electricity instead of making passive revenue.
So, i think i will turn off my rig for next weeks, and i don't know if i continue to use WTM switcher.

I'm asking myself if minning altcoins on zpool or Nicehash in profit switching is a good alternative.

I think papampi should intégrate this in WTM_Switcher like Salfter was did it. We need to make a cost effective comparison, not only a price comparison. And stop minning when rentability is negative.
